augur/utils.py

```diff
@@ -675,19 +691,56 @@ def read_bed_file(bed_file):
     list of int:
         Sorted list of unique zero-indexed sites
     """
-    mask_sites = []
-    try:
-        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
-                          dtype={1:int,2:int}, **PANDAS_READ_CSV_OPTIONS)
-    except ValueError:
-        # Check if we have a header row. Otherwise, just fail.
-        bed = pd.read_csv(bed_file, sep='\t', header=None, usecols=[1,2],
-                          dtype={1:int,2:int}, skiprows=1, **PANDAS_READ_CSV_OPTIONS)
-        print("Skipped row 1 of %s, assuming it is a header." % bed_file)
-    for _, row in bed.iterrows():
-        mask_sites.extend(range(row[1], row[2]))
+    in_header = True
+    initial_chrom_value = ""
+    mask_sites: list[int] = []
+
+    bed_file_size = os.path.getsize(bed_file)
+
+    # Only try to read the file if it is larger than zero bytes;
+    # otherwise return an empty list
+    if bed_file_size == 0:
+        return mask_sites
+
+    with open(bed_file, 'r', newline='') as bed_fh:
+        reader = csv.reader(bed_fh, delimiter='\t')
```

Comment:
I'd think to `.split('\t')` instead of using a CSV-like TSV parser. That parser does way more than what BED is defined as.

Reply:
While the BED spec allows whitespace-separated fields, bedtools requires tabs. I think it's fine to also require tabs since in practice we probably want to be able to use bedtools on these files?